source/dagger/gameplay/team_game/character_controller.cpp

```diff
@@ -46,8 +55,11 @@ void Idle::Run(CharacterController& ctrl_)
 	{
 		GoTo(ECharacterState::Running, ctrl_);
 	}
-
-	AnimatorPlay(*ctrl_.animator, "among_them_animations:knight_idle");
+	if (ctrl_.shapeshiftInput.x == 1) { idle = "among_them_animations:goblin_idle"; running = "among_them_animations:goblin_run"; 	}
+	if (ctrl_.shapeshiftInput.y == 1) { idle = "among_them_animations:slime_idle"; running = "among_them_animations:slime_run";}
+	if (ctrl_.shapeshiftInput.z == 1) { idle = "among_them_animations:bat"; running = "among_them_animations:bat"; }
+	if (ctrl_.shapeshiftInput.w == 1) { idle = "among_them_animations:knight_idle"; running = "among_them_animations:knight_run"; }
```

Comment:
Maybe you can move this to `CharacterControllerSystem::Run`, that way you only write this code once, and you don't need `shapeshiftInput`.
Also, enemies will have to know which shape character has is in to decide whether to start the chase, so that information should be stored somewhere and assigned here.
I really like this solution, though.